packages/kuma-gui/src/app/x/components/x-teleport/XTeleportTemplate.vue

```diff
@@ -7,19 +7,30 @@
   </Teleport>
 </template>
 <script lang="ts" setup>
-import { onMounted, ref } from 'vue'
+import { onMounted, onBeforeUnmount, ref } from 'vue'
 const props = defineProps<{
   to: {
     name: string
   }
 }>()
-
+const controller = new AbortController()
 const ready = ref<boolean>(false)
+
+const isCustomEvent = (e: Event | CustomEvent): e is CustomEvent => 'detail' in e
+
 onMounted(() => {
   if (document.querySelector(`[data-x-teleport-id='${props.to.name}']`) !== null) {
     ready.value = true
   } else {
-    throw new Error(`The '[data-x-teleport-id='${props.to.name}']' element could not be found to teleport to`)
+    window.addEventListener('x-teleport-slot:mounted', (e: Event) => {
+      if (isCustomEvent(e) && e.detail?.name === props.to.name) {
+        ready.value = true
+      }
+    }, { signal: controller.signal })
```

Comment:
Really just a nit and I'll approve anyways, but I think we can save some lines and the type guard here:

```suggestion
    window.addEventListener('x-teleport-slot:mounted', (e: CustomEvent<{ name: string }> | Event) => {
      if ('detail' in e && e.detail?.name === props.to.name) {
        ready.value = true
      }
    }, { signal: controller.signal })
```